eng/mgmt/automation/generate_data.py

```diff
@@ -54,20 +55,30 @@ def sdk_automation_cadl(config: dict) -> List[dict]:
                 module = match.group(2)
 
             if not module:
-                logging.warning('[Skip] sdk-folder option {} not match format sdk/<service>/<module>'
-                                .format(sdk_folder))
+                logging.warning('[Skip] "emitter-output-dir" not match format '
+                                '{java-sdk-folder}/sdk/{service-directory-name}/<module>')
             else:
                 # cadl
                 succeeded = False
                 pwd = os.getcwd()
                 os.chdir(cadl_dir)
                 try:
+                    # install latest @azure-tools/cadl-java
+                    subprocess.run('npm install @azure-tools/cadl-java', shell=True, check=True)
```

Comment:
Currently using latest cadl-java.

Alternative is to fix version of both cadl-java and cadl-autorest. However, this depends on that package.json only include cadl-autorest and nothing else, which might not be the case.